Library/Homebrew/formulary.rb

```diff
@@ -489,17 +491,17 @@ class FromUrlLoader < FormulaLoader
     def initialize(url, from: nil)
       @url = url
       @from = from
-      uri = URI(url)
-      formula = File.basename(uri.path, ".rb")
-      super formula, HOMEBREW_CACHE_FORMULA/File.basename(uri.path)
+      uri_path = T.must(URI(url).path)
+      formula = File.basename(uri_path, ".rb")
+      super formula, HOMEBREW_CACHE_FORMULA/File.basename(uri_path)
     end
 
     def load_file(flags:, ignore_errors:)
       if @from != :formula_installer
-        if %r{githubusercontent.com/[\w-]+/[\w-]+/[a-f0-9]{40}(?:/Formula)?/(?<formula_name>[\w+-.@]+).rb} =~ url
+        if (md = %r{githubusercontent.com/[\w-]+/[\w-]+/[a-f0-9]{40}(?:/Formula)?/(?<name>[\w+-.@]+).rb}.match(url))
```

Comment:
I think we usually use `match` instead of `md` or `match_data`. May make sense to check for other outliers to make this more consistent.

Reply:
Also flip the order here.

Reply:
I can switch the order. The `md` here is an exception for readability, as otherwise this needs to span multiple lines. The regexp should arguably be a const anyway, but I try to minimize the refactoring not directly related to enabling typing (which is also why I preserve order by default).